wasi/wasi.go

```diff
@@ -41,17 +41,18 @@ type FS interface {
 
 // Errno are the error codes returned by WASI functions.
 //
+// Note: This is not always an error, as ErrnoSuccess is a valid code.
 // Note: Codes are defined even when not relevant to WASI for use in higher-level libraries or alignment with POSIX.
 // See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#-errno-enumu16
 // See https://linux.die.net/man/3/errno
-type Errno uint32
+type Errno = uint32 // alias for parity with internalwasm.ValueType
 
-// Error returns the POSIX error code name, except ErrnoSuccess, which isn't defined. Ex. Errno2big -> "E2BIG"
-func (err Errno) Error() string {
-	if int(err) < len(errnoToString) {
-		return errnoToString[err]
+// ErrnoName returns the POSIX error code name, except ErrnoSuccess, which is not an error. Ex. Errno2big -> "E2BIG"
+func ErrnoName(errno Errno) string {
```

Comment:
This change removes golang's `Error` interface from `Errno`. Is it intended? 
By the way, I think it's ok to remove `Error` interface from `Errno` anyway. Considering that in the future we may give host functions the ability to return an `Error` to trigger immediate termination, we will not want `Errno` to become `Error`, because it means even `ErrnoSuccess` triggers immediate termination.

Reply:
yeah this is intended as it isn't an error, as you mentioned. the other benefit is it keeps the function signature easy to implement reflectively (as you don't need to instantiate non-go type)